.github/workflows/main.yaml

```diff
@@ -108,3 +110,20 @@ jobs:
             fi
           done
         working-directory: build
+
+  unit-tests:
+    needs:
+      - setup
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.15
+        id: go
+      - name: Check out code into the Go module directory
```

Comment:
We should probably use a cache for the downloaded go modules to improve performance. For example (this is from the github action's page):

```yaml
- uses: actions/cache@v2
  with:
    path: ~/go/pkg/mod
    key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
    restore-keys: |
      ${{ runner.os }}-go-
```

Reply:
>We should probably use a cache

Yes. Thanks for letting me be lazy and showing me what to add :)